### PR TITLE
Escape sql special like characters

### LINKED ISSUE #902: Make sure query strings and filters cannot be used for SQL attacks
Such as where the run (and product name) filter is accepted from the user.

Also, in these cases with string matching, `_` should match the literal, not the SQL syntax char for "single arbitrary character".

### PR DESCRIPTION
**Pull request by csordasmarton**
Closes #902

### CODE REVIEW

File: libcodechecker/util.py [hunk 1]
```diff
@@ -313,3 +313,10 @@ def sizeof_fmt(num, suffix='B'):
             return "%3.1f%s%s" % (num, unit, suffix)
         num /= 1024.0
     return "%.1f%s%s" % (num, 'Yi', suffix)
+
+
+def escape_like(string, escape_char='*'):
+    """Escape the string parameter used in SQL LIKE expressions."""
+    return string.replace(escape_char, escape_char * 2) \
```
Comment (whisperity) [inline libcodechecker/util.py]: So a `*` becomes `**`? How does this affect the query?

If I have `csordas*marton` as filter and I have `csordas*marton` and `csordasFOOBARmarton` as run names in the database I get only the first as result?

Comment (csordasmarton) [reply libcodechecker/util.py]: If you have two run names (`csordas*marton` and `csordasFOOBARmarton`) you get only the first as result for `csordas*marton` filter.

Comment (bruntib) [reply libcodechecker/util.py]: This escaping applies to the run and product names when the user starts typing an infix of these for filtering. I think it is not necessary to accept joker characters in this situation, so it wouldn't be a problem if asterisk is interpreted as an exact character.

File: tests/functional/report_viewer_api/test_run_filter.py [hunk 2]
```diff
@@ -48,18 +48,23 @@ def test_filter_run_names(self):
...
-        # Filter runs which name contains `files`.
+        # Filter runs which name contains `files_`.
         test_runs = self.__get_runs('files_')
-        self.assertEqual(len(test_runs), 2,
-                         "There should be two runs for this test.")
+        self.assertEqual(len(test_runs), 1,
+                         "There should be one run for this test.")
+
+        # Filter runs which name contains `test_files*`.
+        test_runs = self.__get_runs('test_files*')
```
Comment (Xazax-hun) [inline tests/functional/report_viewer_api/test_run_filter.py]: I think there could be more tests, e.g. for `**, %`.

Comment (csordasmarton) [reply tests/functional/report_viewer_api/test_run_filter.py]: Okay, I've created a few more test cases.

**Pull request merged**